### PR TITLE
141617729 persisting food jc

### PR DESCRIPTION
**Pull request by jdconrad89**
@case-eee @icorson3 @DavidKnott @wlffann After today's class and our conversation from last night around storing the full html in the localStorage I went back and re-did our localStorage setup. We are now storing the food items and their information in string formatted JSON. I've already updated the persisting items on page reloads. Also as I have a feeling we will have to continually plug the information into the table HTML I pulled that process out into its own function so all we will need to do in order to plug the information into the table is call
```formatFoodItemTable(name, calories);``` passing in the necessary information to the function. Let me know if you have any questions!

### CODE REVIEW

File: lib/foods.js [hunk 1]
```diff
@@ -15,66 +15,96 @@ $('document').ready(function() {
...
+
+function add_delete_functionality(){
+  $("td.food-delete button").on('click', function(){
+    $(localStorage.removeItem(this));
+  })
+}
+
+function Food(name, calories) {
+  this.name = name;
+  this.calories = calories;
```
Comment (wlffann) [inline lib/foods.js:70]: 👍

File: lib/foods.js [hunk 2]
```diff
@@ -15,66 +15,96 @@ $('document').ready(function() {
...
+};
+
+function formatFoodItemTable(name, calories) {
+  var foodRow = `<tr class='food-row'>
+  <td class='food-name'>` + name + `</td>
+  <td class='food-calories'>` + calories + `</td>
+  <td class='food-delete'><button>-</button></td>
+  </tr>`
+
+  $("tbody").prepend(foodRow);
```
Comment (wlffann) [inline lib/foods.js]: Love this! I'm going to use it too.

**Review by wlffann — COMMENTED** [review]
Looks great! Thanks for making these changes. This will be much easier to work with

File: lib/foods.js [hunk 3]
```diff
@@ -14,13 +14,7 @@ $('document').ready(function() {
...
     }
 
-    var foodRow = `<tr class='food-row'>
-    <td class='food-name'>` + name + `</td>
-    <td class='food-calories'>` + calories + `</td>
-    <td class='food-delete'><button>-</button></td>
-    </tr>`
-
-    $("tbody").prepend(foodRow);
+    formatFoodItemTable(name, calories);
```
Comment (DavidKnott) [inline lib/foods.js]: Great job extracting this into a method!

**Comment by DavidKnott** [comment]
Looks great, after we merge in my PR would you want to squash some of your smaller commits?

**Comment by DavidKnott** [comment]
Looks solid, I feel comfortable merging this

**Comment by case-eee** [comment]
@jdconrad89 @DavidKnott @wlffann be very careful of your syntax! you're mixing snake case with camel case - i highly suggest fixing this before merging 👍

**Comment by jdconrad89** [comment]
@case-eee the naming conventions have been fixed!

**Pull request merged**